pyiceberg/io/pyarrow.py

```diff
@@ -352,14 +352,21 @@ def parse_location(location: str) -> Tuple[str, str, str]:
 
     def _initialize_fs(self, scheme: str, netloc: Optional[str] = None) -> FileSystem:
         if scheme in {"s3", "s3a", "s3n", "oss"}:
-            from pyarrow.fs import S3FileSystem
+            from pyarrow.fs import S3FileSystem, resolve_s3_region
+
+            bucket_region = None
+            if netloc:
```

Comment:
nit: add a comment here about automatically resolving the region from a bucket name.